pkg/monitor/api.go

```diff
@@ -182,6 +193,117 @@ func startServerMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.
 	return nil
 }
 
+func startNetworkMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.Config, timeout time.Duration, resourceLocator, url string, disableConnectionReuse bool) error {
+	kubeTransportConfig, err := clusterConfig.TransportConfig()
+	if err != nil {
+		return err
+	}
+	tlsConfig, err := transport.TLSConfigFor(kubeTransportConfig)
+	if err != nil {
+		return err
+	}
+	var httpTransport *http.Transport
+
+	if disableConnectionReuse {
+		httpTransport = &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   timeout,
+				KeepAlive: -1, // this looks unnecessary to me, but it was set in other code.
```

Comment:
I will keep this, it will detect a connection hanging , it is a TCP option and vI think desirable
https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/net/dial.go;l=73-80